File: code/typescript/src/iterator/BlogPlatform.ts
import { Blog } from "./Blog";
import { Post } from "./Post";
import { PostIterator } from "./PostIterator";

export class BlogPlatform implements Blog {
    private posts: Post[] = [];

    addPost(post: Post): void {
        this.posts.push(post);
    }

    createIterator(): PostIterator {
        return new PostIterator(this.posts);
    }
}

File: code/python/iterator/BlogPlatform.py
from Blog import Blog
from PostIterator import PostIterator


class BlogPlatform(Blog):
    def __init__(self):
        self.posts = []

    def add_post(self, post):
        self.posts.append(post)

    def create_iterator(self):
        return PostIterator(self.posts)
